docs/DeveloperGuide.adoc

```diff
@@ -153,98 +173,236 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
-The following sequence diagram shows how the undo operation works:
+image::DueSoonActivityDiagram.png[width = "400", length = "600"]
 
-image::UndoSequenceDiagram.png[]
+Sequence diagram:
 
-NOTE: The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+image::DueSoonSequenceDiagram.png[width = "600", length = "500"]
 
-The `redo` command does the opposite -- it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+==== Design Considerations
+
+===== Aspect: How `due soon` command works
+
+* **Alternative 1 (current choice):** Does not clear and reload the due soon task list.
+** Pros: Fewer operations need to be performed.
+** Cons: May have be more prone to errors because list is not completely "refreshed". Some parts of it are the same.
+* **Alternative 2:** Clear and reload the due soon task list.
+** Pros: Less prone to errors since the entire list is "refreshed".
+** Cons: More operation intensive.
+// end::dueSoon[]
+
+// tag::taskSummary[]
+=== Task Summary Feature
+
+==== Implementation
+
+The task summary feature allows user to enjoy visualized and real-time statistics of their tasks' information.
+
+This feature is implemented using different charts as `pie chart`, `area chart`, `bar chart` and `line chart`.
+All charts are organized in a `TabPanel` which will replace the `ListPanel` and `Due Soon List Panel` in the main window when
+`Task Summary` in the `Menu` is clicked.
+
+The data of each chart will automatically rebind when user manage their tasks using the commands as below:
+
+* `add`
+* `done`
+* `delete`
+* `refresh`
 
 [NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+Only when there is task status change after `refresh`, the charts will refresh accordingly.
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+==== Example Usage Scenario
+Given below is an example usage scenario and how the task summary mechanism behaves at each step.
 
-image::UndoRedoState4.png[]
+Step 1. The user launches the application with some alive and archived task records in the `storage`.
+Statistics will perform on the records retrieved, and the result will be visualized as different charts organized in a `tab panel`.
+After the user clicks `Statistics -> Task Summary`, the `tab panel` will display.
```

Comment:
This might not be complying to the project requirement of CLI? is there a CLI command for this?